overlord/snapstate/target.go

```diff
@@ -104,10 +104,10 @@ type StoreSnap struct {
 	SkipIfPresent bool
 }
 
-// verify that StoreTarget implements the Target interface
-var _ InstallGoal = &storeInstallGoal{}
+// verify that storeInstallGoal implements the installGoal interface
+var _ installGoal = &storeInstallGoal{}
 
-// StoreGoal creates a new storeInstallGoal, which implements InstallGoal to
+// StoreGoal creates a new storeInstallGoal, which implements installGoal to
 // install snaps from the store.
 func StoreGoal(snaps ...StoreSnap) *storeInstallGoal {
```

Comment:
this should use the interface as return type? same for the other Goal constructor